app/dao/jobs_dao.py

```diff
@@ -44,6 +44,17 @@ def dao_get_scheduled_jobs():
         .all()
 
 
+def dao_get_future_scheduled_job_by_id_and_service_id(job_id, service_id):
+    return Job.query \
+        .filter(
+            Job.service_id == service_id,
+            Job.id == job_id,
+            Job.job_status == 'scheduled',
```

Comment:
probs best to use the static var to save on magic strings - `from app.models import JOB_STATUS_SCHEDULED`